tests/test_client.py

```diff
@@ -80,6 +80,35 @@ def test_client_authentication_url(api_client, api_url):
     assert urls_equal(expected3, actual3)
 
 
+def test_client_authentication_url_custom_scopes(api_client_scopes, api_url):
+    expected = (
+        URLObject(api_url)
+        .with_path("/oauth/authorize")
+        .set_query_params([
+            ('login_hint', ''),
+            ('state', ''),
+            ('redirect_uri', '/redirect'),
+            ('response_type', 'code'),
+            ('client_id', 'None'),
+            ('scopes', 'email,calendars,contacts'),
```

Comment:
It might also be good to handle the no-scopes case and maybe a 1-scope case (like only calendar).

Also I think it's `calendar` (singular) not `calendars`

Reply:
You're right, it's `calendar` -- fixed.

Do we want to allow the developer to set no scopes? Right now, the codebase will treat that the same way as though the developer didn't pass a `scopes` argument at all, and so it will use the default of all scopes.